src/Util/NamespacesHelper.php

```diff
@@ -0,0 +1,243 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Symfony MakerBundle package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\MakerBundle\Util;
+
+final class NamespacesHelper
+{
+    /** @var string */
+    private $command;
+
+    /** @var string */
+    private $controller;
+
+    /** @var string */
+    private $entity;
+
+    /** @var string */
+    private $fixtures;
+
+    /** @var string */
+    private $form;
+
+    /** @var string */
+    private $functionalTest;
+
+    /** @var string */
+    private $repository;
+
+    /** @var string */
+    private $root;
+
+    /** @var string */
+    private $security;
+
+    /** @var string */
+    private $serializer;
+
+    /** @var string */
+    private $subscriber;
+
+    /** @var string */
+    private $twig;
+
+    /** @var string */
+    private $unitTest;
+
+    /** @var string */
+    private $validator;
+
+    public function __construct(
+        string $root,
+        string $command = null,
+        string $controller = null,
+        string $entity = null,
+        string $fixtures = null,
+        string $form = null,
+        string $functionalTest = null,
+        string $repository = null,
+        string $security = null,
+        string $serializer = null,
+        string $subscriber = null,
+        string $twig = null,
+        string $unitTest = null,
+        string $validator = null
+    ) {
+        $this->root = $root;
+        $this->command = $command ?? 'Command\\';
+        $this->controller = $controller ?? 'Controller\\';
+        $this->entity = $entity ?? 'Entity\\';
+        $this->fixtures = $fixtures ?? 'DataFixtures\\';
+        $this->form = $form ?? 'Form\\';
+        $this->functionalTest = $functionalTest ?? 'Tests\\';
+        $this->repository = $repository ?? 'Repository\\';
+        $this->security = $security ?? 'Security\\';
+        $this->serializer = $serializer ?? 'Serializer\\';
+        $this->subscriber = $subscriber ?? 'EventSubscriber\\';
+        $this->twig = $twig ?? 'Twig\\';
+        $this->unitTest = $unitTest ?? 'Tests\\';
+        $this->validator = $validator ?? 'Validator\\';
+    }
```

Comment:
It's going to be weird either way, but I think we should just accept a `$namespaces` array, then do a check to make sure all the "keys" we expect are there (and throw an exception if any are missing or extra). I don't want to keep addming more and more arguments as we generate more and more stuff. But having the individual getters is ok.

Reply:
@weaverryan Do we really need to check the given namespaces? 

- If a namespace isn't given we will default it and doing it in the helper directly has the advantage of having a single place to define the default value of a namespace.

- If an extra namespace is given it won't have any impact because as you said we have the individual getters which look for specific namespaces so there will be no way to get an extra namespace from the namespaces helper.

But I completely agree with your point about keep adding more arguments each time we want to introduce a new namespace, so I've implemented a mix of our 2 solutions, please have a look and let me know what you think.